superbench/common/utils/nvidia_helper.py

```diff
@@ -5,18 +5,180 @@
+        util = nvml.nvmlDeviceGetUtilizationRates(handle)
+    except nvml.NVMLError as err:
+        logger.error('Get device utilization failed: {}'.format(str(err)))
+        return None
+    return util.gpu
+
+
+def get_device_temperature(idx):
+    """Get the temperature of device, unit: celsius.
+
+    Args:
+        idx (int): device index.
+
+    Return:
+        temp (int): the temperature of device, None means failed to get the data.
+    """
+    try:
+        handle = nvml.nvmlDeviceGetHandleByIndex(idx)
+        temp = nvml.nvmlDeviceGetTemperature(handle, nvml.NVML_TEMPERATURE_GPU)
+    except nvml.NVMLError as err:
+        logger.error('Get device temperature failed: {}'.format(str(err)))
+        temp = None
+    return temp
+
+
+def get_device_power_limit(idx):
+    """Get the power management limit of device, unit: watt.
+
+    Args:
+        idx (int): device index.
+
+    Return:
+        temp (float): the power management limit of device, None means failed to get the data.
+    """
+    try:
+        handle = nvml.nvmlDeviceGetHandleByIndex(idx)
+        powerlimit = nvml.nvmlDeviceGetPowerManagementLimit(handle)
+    except nvml.NVMLError as err:
+        logger.error('Get device power limitation failed: {}'.format(str(err)))
+        return None
+    return int(int(powerlimit) / 1000)
+
+
+def get_device_memory(idx):
+    """Get the memory information of device, unit: byte.
+
+    Args:
+        idx (int): device index.
+
+    Return:
+        used (float): the used device memory, None means failed to get the data.
+        total (float): the total device memory, None means failed to get the data.
+    """
+    try:
+        handle = nvml.nvmlDeviceGetHandleByIndex(idx)
+        mem = nvml.nvmlDeviceGetMemoryInfo(handle)
+    except nvml.NVMLError as err:
+        logger.error('Get device memory failed: {}'.format(str(err)))
+        return None, None
+    return mem.used, mem.total
+
+
+def get_device_row_remapped_info(idx):
+    """Get the row remapped information of device.
+
+    The command 'nvidia-smi -i idx -q' contains the following output:
+        Remapped Rows
+            Correctable Error                 : 0
+            Uncorrectable Error               : 0
+            Pending                           : No
+            Remapping Failure Occurred        : No
+            Bank Remap Availability Histogram
+                Max                           : 640 bank(s)
+                High                          : 0 bank(s)
+                Partial                       : 0 bank(s)
+                Low                           : 0 bank(s)
+                None                          : 0 bank(s)
+        Temperature
+            GPU Current Temp                  : 36 C
+
+    Args:
+        idx (int): device index.
+
+    Return:
+        remapped_metrics (dict): the row remapped information, None means failed to get the data.
+    """
+    output = run_command('nvidia-smi -i {} -q'.format(idx))
```

Comment:
why not use `nvmlDeviceGetRemappedRows`?

Reply:
`nvmlDeviceGetRemappedRows` doesn't have python API support.